src/overlays/actors/ovl_En_GeldB/z_en_geldb.c

```diff
@@ -33,93 +97,1466 @@ const ActorInit En_GeldB_InitVars = {
+                ((globalCtx->gameplayFrames & 7) || (ABS(angleToFacingLink) < 0x38E0))) {
+                EnGeldB_SetupSlash(this);
+                return true;
+            } else {
+                EnGeldB_SetupCircle(this);
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+void EnGeldB_SetupWait(EnGeldB* this) {
+    SkelAnime_ChangeAnimPlaybackStop(&this->skelAnime, &D_0600ADF8, 0.0f);
+    this->actor.posRot.pos.y = this->actor.initPosRot.pos.y + 120.0f;
+    this->timer = 10;
+    this->invisible = true;
+    this->actionState = GELDB_WAIT;
+    this->actor.bgCheckFlags &= ~3;
+    this->actor.gravity = -2.0f;
+    this->actor.flags &= ~1;
+    EnGeldB_SetupAction(this, EnGeldB_Wait);
+}
+
+void EnGeldB_Wait(EnGeldB* this, GlobalContext* globalCtx) {
+    if ((this->invisible && !Flags_GetSwitch(globalCtx, this->actor.initPosRot.rot.z)) ||
+        this->actor.xzDistFromLink > 300.0f) {
+        this->actor.shape.rot.y = this->actor.posRot.rot.y = this->actor.yawTowardsLink;
+        this->actor.posRot.pos.y = this->actor.groundY + 120.0f;
+    } else {
+        this->invisible = false;
+        this->actor.shape.unk_10 = 90.0f;
+        func_800F5ACC(0x38);
+    }
+    if (this->actor.bgCheckFlags & 2) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIZA_DOWN);
+        this->skelAnime.animPlaybackSpeed = 1.0f;
+        this->actor.posRot.pos.y = this->actor.groundY;
+        this->actor.flags |= 1;
+        this->actor.posRot2.pos = this->actor.posRot.pos;
+        this->actor.bgCheckFlags &= ~2;
+        this->actor.velocity.y = 0.0f;
+        func_80033260(globalCtx, &this->actor, &this->rightFootPos, 3.0f, 2, 2.0f, 0, 0, 0);
+        func_80033260(globalCtx, &this->actor, &this->leftFootPos, 3.0f, 2, 2.0f, 0, 0, 0);
+    }
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        EnGeldB_SetupReady(this);
+    }
+}
+
+void EnGeldB_SetupFlee(EnGeldB* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600ADF8, -2.0f, SkelAnime_GetFrameCount(&D_0600ADF8), 0.0f, 3, -4.0f);
+    this->timer = 20;
+    this->invisible = false;
+    this->actionState = GELDB_WAIT;
+    this->actor.shape.rot.y = this->actor.posRot.rot.y = this->actor.yawTowardsLink;
+    EnGeldB_SetupAction(this, EnGeldB_Flee);
+}
+
+void EnGeldB_Flee(EnGeldB* this, GlobalContext* globalCtx) {
+    if (this->skelAnime.animCurrentFrame == 10.0f) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_JUMP);
+    }
+    if (this->skelAnime.animCurrentFrame == 2.0f) {
+        this->actor.gravity = 0.0f;
+        func_80033260(globalCtx, &this->actor, &this->rightFootPos, 3.0f, 2, 2.0f, 0, 0, 0);
+        func_80033260(globalCtx, &this->actor, &this->leftFootPos, 3.0f, 2, 2.0f, 0, 0, 0);
+    }
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.y, this->actor.groundY + 300.0f, 1.0f, 20.5f, 0.0f);
+        if (--this->timer == 0) {
```

Comment:
i prefer decrement seperate from the check vs pre dec, but not a requirement. just putting it out there

Reply:
I do too, but it has to be used elsewhere for the else ifs, so I was trying to be consistent within the file. If we don't care about that, I can pull them out.

Reply:
ah yea i noticed that too. I would personally not do it unless necessary like in those other cases, but just a personal preference

Reply:
pulled them out of the ifs